### ISSUE TITLE
Amazon Kindle embed wrongly treats products as previewable e-books

### ISSUE THREAD

**Issue by mcsf** [WordPress/gutenberg#27547]
Likely introduced in #13510.
Reported in https://github.com/WordPress/gutenberg/pull/13510/files#r537489307

**Describe the bug**

When pasting Amazon URLs, such as for products that aren't e-books, the editor inserts an Amazon Kindle embed regardless of the resource that the URL describes.

**To reproduce**

1. Grab a Amazon URL describing something that isn't an e-book, such as this vinyl: `https://www.amazon.com/dp/B0198CGQEE/`
2. Paste the URL into the editor
3. Notice the insertion of an Amazon Kindle embed block
4. Try interacting with the block (either in the editor or the front end) and note how it wrongly presents a Preview button and how the Share function offers an invalid shortlink.

**Expected behavior**

* The URL obtained via the embed's Share button should work
* There should be no "Free Preview" button

**Screenshots**

![output](https://user-images.githubusercontent.com/150562/101354667-69b5d300-388d-11eb-83dd-b39c7b52a297.gif)

**Comment by mtias** [comment]
Can we remove the kindle block if this isn't addressed or is too fragile?

**Comment by mcsf** [comment]
I don't think that really changes things, since this all comes from the embed endpoint. The Kindle block is just an explicit block variation of the Embed block. Follow-up is more likely needed in core:

* Enacted in [Changeset 41615: Embeds: Add support for Amazon Kindle instant previews](https://core.trac.wordpress.org/changeset/41615)
* Tracked in [Ticket 38181: Add oEmbed support for Amazon Kindle Instant Previews](https://core.trac.wordpress.org/ticket/38181)

cc @Soean: perhaps you have a preference, since you introduced the block in #13510.

**Comment by SilverNightFall** [comment]
Yes I got this message too when posting a link in the message section of BuddyBoss. I submitted a ticket and they verified it was a core issue with WordPress.  I shared an Amazon Link

![Image](https://github.com/user-attachments/assets/995b381b-9e48-443a-86bf-ccc77f4d893e)

**Comment by youknowriad** [comment]
Closing as the related trac tickets seem to have been fixed.
